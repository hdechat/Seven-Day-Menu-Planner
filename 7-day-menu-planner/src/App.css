body {
  background-repeat: no-repeat;
  box-sizing: border-box;
}

h1 {
}

.App {
  text-align: center;
}

.App-header {
  background: #878984;
  display: flex;
  justify-content: space-around;
  height: 2rem;
  padding: 0rem 30rem 0.5rem 20rem;
}

.top-of-nav {
  margin: -2px;
}

.bottom-of-nav {
  margin: 6px;
}

.nav {
  align-self: baseline;
  color: #fff;
  font-size: 1.2rem;
  margin: 0;
  padding: 0.5rem 3rem;
  text-decoration: none;
}

.nav:hover {
  color: #9bd437;
}

.title {
  background: #878984;
  color: #9bd437;
  text-decoration: none;
  float: left;
  font-size: 1.5rem;
  margin: 0;
  padding-left: 5px;
  line-height: 39px;
}

.welcome-message {
  color: #de3400;
  font-size: 50px;
}

@media screen and (max-width: 350px) {
  body {
    background: #ADC3D0;
  }

  .App-header {
    height: 5rem;
    margin-top: -1.5rem;
    width: fit-content;
  }

  .title {
    display: none;
  }

  .nav {
    margin: 2rem 1.7rem 2rem 2rem;
    font-size: 1.5rem;
  }

  .english-spanish {
      margin: auto 17rem;
  }

  .select-category {
    height: 3rem;
    margin: 1rem;
    width: 14rem;
  }

  .search {
    font-size: 20px;
    width: 14rem;
    height: 3rem;
  }

  .search::placeholder {
    font-size: 20px;
  }

  .welcome-message {
    font-size: 40px;
    width: 36rem;
  }

  .display-cards {
    overflow-y: unset;
    width: 180%;
  }

  select {
    font-size: 20px;
    height: 3rem;
  }
}
