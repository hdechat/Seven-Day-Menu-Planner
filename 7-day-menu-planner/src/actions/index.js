export const chooseCategory = (category) => ({
  type: 'CHOOSE_CATEGORY',
  category
});

export const fetchRecipesSuccess = (recipes) => ({
  type: 'FETCH_RECIPES_SUCCESS',
  recipes
});

export const recipesIsLoading = (bool) => ({
  type: 'RECIPES_IS_LOADING',
  recipesIsLoading: bool
});

export const recipesHasErrored = (error) => ({
  type: 'RECIPES_HAS_ERRORED',
  error
});

export const addToMenu = (weekday, mealTime, recipe) => ({
  type: 'ADD_TO_MENU',
  weekday,
  mealTime,
  recipe
});

export const removeFromMenu = (weekday, mealTime) => ({
  type: 'REMOVE_FROM_MENU',
  weekday,
  mealTime
});

export const addToGroceryList = (ingredients) => ({
  type: 'ADD_TO_GROCERY_LIST',
  ingredients
});

export const removeFromGroceryList = (ingredients) => ({
  type: 'REMOVE_FROM_GROCERY_LIST',
  ingredients
});

export const toggleFavorites = (toggledRecipe) => ({
  type: 'TOGGLE_FAVORITES',
  toggledRecipe
});

//LOCAL STORAGE ACTIONS
export const retrieveDataFromStorage = (key) => ({
  type: 'RETRIEVE_DATA_FROM_STORAGE',
  key
});

export const loadFavoritesToStore = (favorites) => ({
  type: 'LOAD_FAVORITES_TO_STORE',
  favorites
});

export const loadGroceryListToStore = (groceryList) => ({
  type: 'LOAD_GROCERY_LIST_TO_STORE',
  groceryList
});

export const loadMenuToStore = (menu) => ({
  type: 'LOAD_MENU_TO_STORE',
  menu
});

export const addFavoriteToStorage = (recipe) => ({
  type: 'ADD_FAVORITE_TO_STORAGE',
  recipe
});

export const removeFavoriteFromStorage = (recipe) => ({
  type: 'REMOVE_FAVORITE_FROM_STORAGE',
  recipe
});

export const addGroceryListToStorage = (groceryList) => ({
  type: 'ADD_GROCERY_LIST_TO_STORAGE',
  groceryList
});

export const removeGroceryListFromStorage = (groceryList) => ({
  type: 'REMOVE_GROCERY_LIST_FROM_STORAGE',
  groceryList
});

export const addMenuItemToStorage = (weekday, mealTime, recipe) => ({
  type: 'ADD_MENU_ITEM_TO_STORAGE',
  weekday,
  mealTime,
  recipe
});

export const removeMenuItemFromStorage = (weekday, mealTime) => ({
  type: 'REMOVE_MENU_ITEM_FROM_STORAGE',
  weekday,
  mealTime,
});