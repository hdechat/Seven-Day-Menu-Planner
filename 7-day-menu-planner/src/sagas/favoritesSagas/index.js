import { call, put, takeEvery  } from 'redux-saga/effects';
import { setToStorage, getDataFromStorage } from '../../api';

export function* addFavoriteToStorage(action) {
  let updatedFavorites;
  const favorites = yield call(getDataFromStorage, 'favorites');
  favorites 
    ? updatedFavorites = [...favorites, action.recipe]
    : updatedFavorites = [action.recipe];
  yield call(setToStorage, 'favorites', updatedFavorites);
}

export function* removeFavoriteFromStorage(action) {
  const favorites = yield call(getDataFromStorage, 'favorites');
  const updatedFavorites = favorites.filter(recipe => 
    recipe.title !== action.recipe.title)
  yield call(setToStorage, 'favorites', updatedFavorites);
}

export function* listenForAddFavoriteToStorage() {
  yield takeEvery('ADD_FAVORITE_TO_STORAGE', addFavoriteToStorage);
}

export function* listenForRemoveFavoriteFromStorage() {
  yield takeEvery('REMOVE_FAVORITE_FROM_STORAGE', removeFavoriteFromStorage);
}
