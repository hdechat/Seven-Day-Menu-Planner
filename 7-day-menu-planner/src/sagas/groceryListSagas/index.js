import { call, put, takeEvery } from 'redux-saga/effects';
import { setToStorage, getDataFromStorage } from '../../api';

export function* addGroceryListToStorage(action) {
  let updatedGroceryList;
  const groceryList = yield call(getDataFromStorage, 'groceryList');
  groceryList
    ? updatedGroceryList = [...groceryList, ...action.groceryList]
    : updatedGroceryList = action.groceryList;
  yield call(setToStorage, 'groceryList', updatedGroceryList);
}

export function* removeGroceryListFromStorage(action) {
  const groceryList = yield call(getDataFromStorage, 'groceryList');
  const updatedGroceryList = groceryList.filter(ingred => 
    !action.groceryList.includes(ingred));
  yield call(setToStorage, 'groceryList', updatedGroceryList);
}

export function* listenForAddGroceryListToStorage() {
  yield takeEvery('ADD_GROCERY_LIST_TO_STORAGE', addGroceryListToStorage);
}

export function* listenForRemoveGroceryListFromStorage() {
  yield takeEvery('REMOVE_GROCERY_LIST_FROM_STORAGE', removeGroceryListFromStorage);
}