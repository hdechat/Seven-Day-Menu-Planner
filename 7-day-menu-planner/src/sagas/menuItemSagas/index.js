import { call, put, takeEvery } from 'redux-saga/effects';
import { setToStorage, getDataFromStorage } from '../../api';

export function* addMenuItemToStorage(action) {
  let updatedMenu;
  const menu = yield call(getDataFromStorage, 'menu');
  menu
    ? updatedMenu = 
      [...menu, { [action.weekday]: {[action.mealTime]: action.recipe} }]
    : updatedMenu = 
      [{ [action.weekday]: {[action.mealTime]: action.recipe} }];
  yield call(setToStorage, 'menu', updatedMenu)
}

export function* removeMenuItemFromStorage(action) {
  const menu = yield call(getDataFromStorage, 'menu');
  const updatedMenu = menu.filter(item => {
    return JSON.stringify(item) !== 
      JSON.stringify({ [action.weekday]: {[action.mealTime]: action.recipe} })
  });
  yield call(setToStorage, 'menu', updatedMenu);
}


export function* listenForAddMenuItemToStorage() {
  yield takeEvery('ADD_MENU_ITEM_TO_STORAGE', addMenuItemToStorage);
}

export function* listenForRemoveMenuItemFromStorage() {
  yield takeEvery('REMOVE_MENU_ITEM_FROM_STORAGE', removeMenuItemFromStorage);
}
