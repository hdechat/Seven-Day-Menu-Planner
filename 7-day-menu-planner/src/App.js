import React, { Component } from 'react';
import './App.css';
import { NavLink, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { chooseCategory } from './actions';
import RecipeCardsDisplay from './components/RecipeCardsDisplay';
import MenuCalendar from './components/MenuCalendar';

export class App extends Component {

  displayLoadingMessage() {
    return this.props.recipesIsLoading ?
      <p className="loading-message">Your Recipes Are Loading</p> : null;
  }

  render() {
    const { recipeCards, chooseCategory } = this.props;

    return (
      <div className="App">
        <header>
          <NavLink to='/menu-calendar'>Menu Calendar</NavLink>
          <Route exact path='/menu-calendar' component={MenuCalendar} />
        </header>
        <select 
          onChange={event => chooseCategory(event.target.value)}
          id='category'>
          <option>SELECT CATEGORY</option>
          <option value='paleo'>Paleo</option>
          <option value='vegetarian'>Vegetarian</option>
          <option value='vegan'>Vegan</option>
        </select>
        {this.displayLoadingMessage()}
        <RecipeCardsDisplay recipeCards={recipeCards} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  recipesIsLoading: state.recipes.recipesIsLoading,
  recipeCards: state.recipes.results
});
export const mapDispatchToProps = dispatch => ({
  chooseCategory: category => dispatch(chooseCategory(category))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
