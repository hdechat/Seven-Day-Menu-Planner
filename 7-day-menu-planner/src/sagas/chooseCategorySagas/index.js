import { call, put, takeLatest, takeEvery } from 'redux-saga/effects';
import fetchRecipes from '../../api';
import * as actions from '../../actions';
import { cleanData } from '../../helpers';
import { paleo } from '../../mock-data.js';

export function* chooseCategory(action) {
  try {
    yield put(actions.recipesIsLoading(true));
    // const results = paleo;
    const results = yield call(fetchRecipes, action.lang, action.category);
    const recipes = cleanData(results);
    yield put(actions.fetchRecipesSuccess(recipes));
    yield put(actions.recipesIsLoading(false));
  } catch (error) {
    yield put(actions.recipesHasErrored(error.message));
  }
}

export function* listenForChooseCategory() {
  yield takeLatest('CHOOSE_CATEGORY', chooseCategory);
}