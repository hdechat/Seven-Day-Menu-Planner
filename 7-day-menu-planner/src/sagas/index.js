import { call, put, takeLatest } from 'redux-saga/effects';
import { fetchRecipes } from '../api';
import { chooseCategory } from '../actions';

export function* chooseCategory(action) {
  try {
    const data = yield call(fetchRecipes, action.category)
  }
}









// export function* submitUserLogin(action) {
//   try {
//     const user = yield call(API.postLoginUser, action.email, action.password)
//     yield put(actions.loginUser(user))
//   } catch(error) {
//     yield put(actions.loginError(error.message))
//   }
// }

// export function* listenForSubmitUserLogin() {
//   yield takeLatest('SUBMIT_USER_LOGIN', submitUserLogin)
// }

// export default listenForSubmitUserLogi