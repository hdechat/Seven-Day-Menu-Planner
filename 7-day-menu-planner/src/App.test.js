import React from 'react';
import ReactDOM from 'react-dom';
import { App } from './App';
import { shallow, mount } from 'enzyme';
import { chooseCategory } from './actions';
import { mapStateToProps, mapDispatchToProps } from './App.js';
import * as sagas from './sagas';

describe('App', () => {
  let wrapper;
  let mockFunction;

  beforeEach(() => {
    mockFunction = jest.fn()

    wrapper = shallow(<App chooseCategory={mockFunction} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  //***need to figure out how to get around this error: 
  //Invariant Violation: Could not find "store" in either the context or props of "Connect(AddToMenu)".
  //Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(AddToMenu)".
  // it.only('should call handleClick when user selects category', () => {
  //   wrapper = mount(<App chooseCategory={mockFunction} recipeCards={ [{title: 'Recipe Name'}] }/>);

  //   const spy = spyOn(wrapper.instance(), 'handleClick');
    
  //   wrapper.instance().forceUpdate();
  //   wrapper.find('select').simulate('change', {target: {value: 'paleo'}});

  //   expect(spy).toHaveBeenCalled();
  // });

  it('should upate category in state when category is selected', async () => {
    const mockEvent = { target: {value: 'paleo'} };

    await wrapper.instance().handleClick(mockEvent);

    expect(wrapper.state('category')).toEqual('paleo')
  });

  it('should call chooseCategory with the selected category when handleClick is fired', async () => {
    const mockEvent = { target: {value: 'paleo'} };

    await wrapper.instance().handleClick(mockEvent);

    expect(mockFunction).toHaveBeenCalledWith('paleo');
  });
  
  it('should call dispatch with a chooseCategory action', () => {
    const dispatch = jest.fn();
    const actionToDispatch = chooseCategory('paleo');
    const mappedProps = mapDispatchToProps(dispatch);

    mappedProps.chooseCategory('paleo');

    expect(dispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should return the prop object', () => {
    const mockState = {
      recipes: {
        results: [{title: 'Recipe Name'}],
        recipesIsLoading: false,
        recipesHasErrored: false
      }
    };

    const expected = { 
      recipesIsLoading: false, 
      recipeCards: [{title: 'Recipe Name'}] 
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected);
  });

  it('should render message if recipesIsLoading is true', () => {
    expect(wrapper.find('.loading-message').length).toEqual(0);    

    wrapper.setProps({recipesIsLoading: true});

    expect(wrapper.find('.loading-message').length).toEqual(1);
  });
});
