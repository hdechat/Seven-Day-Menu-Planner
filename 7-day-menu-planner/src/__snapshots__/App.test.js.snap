// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        id="diet-plan"
>
        <option
                value="paleo"
        >
                Paleo
        </option>
        <option
                value="vegetarian"
        >
                Vegetarian
        </option>
        <option
                value="vegan"
        >
                Vegan
        </option>
</select>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="paleo"
>
            Paleo
</option>,
          <option
            value="vegetarian"
>
            Vegetarian
</option>,
          <option
            value="vegan"
>
            Vegan
</option>,
        ],
        "id": "diet-plan",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Paleo",
            "value": "paleo",
          },
          "ref": null,
          "rendered": "Paleo",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vegetarian",
            "value": "vegetarian",
          },
          "ref": null,
          "rendered": "Vegetarian",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vegan",
            "value": "vegan",
          },
          "ref": null,
          "rendered": "Vegan",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          id="diet-plan"
>
          <option
                    value="paleo"
          >
                    Paleo
          </option>
          <option
                    value="vegetarian"
          >
                    Vegetarian
          </option>
          <option
                    value="vegan"
          >
                    Vegan
          </option>
</select>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="paleo"
>
              Paleo
</option>,
            <option
              value="vegetarian"
>
              Vegetarian
</option>,
            <option
              value="vegan"
>
              Vegan
</option>,
          ],
          "id": "diet-plan",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Paleo",
              "value": "paleo",
            },
            "ref": null,
            "rendered": "Paleo",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vegetarian",
              "value": "vegetarian",
            },
            "ref": null,
            "rendered": "Vegetarian",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vegan",
              "value": "vegan",
            },
            "ref": null,
            "rendered": "Vegan",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
